### ISSUE TITLE
Error Upload KML extension

### ISSUE THREAD

**Issue by achorivai** [GeoNode/geonode#6080]
## Expected Behavior
Upload and Download KML Extension Featured

## Actual Behavior
![image](https://user-images.githubusercontent.com/4677202/81776800-adc03f80-9519-11ea-9731-b38da87a427c.png)

Client Error: Bad Request
Unexpected exception 'bytes' object has no attribute 'encode'

## Steps to Reproduce the Problem

  1.Upload KML Extension file
  2.There is no option for KML to download layers
  3.

## Specifications

  - GeoNode version: 3.0 Dev
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker):Manual
  - Platform: Ubuntu 18.04
  - Additional details:

**Comment by achorivai** [comment]
After follow the changed from https://github.com/GeoNode/geonode/commit/cbcd06bc34ed1e792763535085ab284dafc80228 there is another error

here is my capture
![image](https://user-images.githubusercontent.com/4677202/81968587-c8440700-9646-11ea-86fa-fd382e43609b.png)

Any suggest please?


fyi, upload shp or geotiff working good and normal (error only for KML files)

**Comment by afabiani** [comment]
any chance to get this KML file?

Also, can you try to upload it on https://dev.geonode.geo-solutions.it/

**Comment by achorivai** [comment]
hi @afabiani 

i already try to upload to http://dev.geonode.geo-solusionts.it/ and same error like my project, here is my captured
![image](https://user-images.githubusercontent.com/4677202/82109728-8b6d3280-9762-11ea-9fb0-d54934e0c2df.png)

yes of course you can try this KML file
[fitur_pal_batas.kml.zip](https://github.com/GeoNode/geonode/files/4637782/fitur_pal_batas.kml.zip)

**Comment by afabiani** [comment]
@achorivai so I taken a look at the KML import.
At a preliminary analysis I discovered that:

- Your KML is not valid for GeoServer. It contains a folder structure. I'm afraid GeoServer Importer is not able to read such complex KMLs. By trying to import it directly on GeoServer, the import fails anyway.

- I tried some simple KMLs, which look to be valid and correctly imported into GeoServer, but in the end GeoServer always produced empty layers.

So, long story short, I guess that at this stage GeoServer Importer has severe issues with KML formats unfortunately.

This is not something the depends on GeoNode. We will need to open a ticket on GeoServer JIRA and look for someone to take a look.

**Comment by achorivai** [comment]
ok @afabiani, thanks for you help 
i will try export KML file from other old version Geonode and upload to this new Geonode 3.0 Dev
